### DISCUSSION TITLE
usage/simple-utilities/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#16988] [Docs]
# usage/simple-utilities/

Explore essential utilities in the Ultralytics package to speed up and enhance your workflows. Learn about data processing, annotations, conversions, and more.

https://docs.ultralytics.com/usage/simple-utilities/

**Comment by JustAnalyze** [comment]
How can I save only specific annotations, like autoannotating just dogs and only getting bounding boxes instead of segmentations?

**Reply by glenn-jocher** [reply]
@JustAnalyze to save only specific annotations like dogs and get bounding boxes instead of segmentations, use the `auto_annotate` function for initial annotation file
    with open(annotations_file, 'r') as f:
        annotations = json.load(f)

    # Filter annotations for the specified class
    filtered_annotations = [ann for ann in annotations if ann['category_id'] == class_id]

    # Create output directory if it doesn't exist
    os.makedirs(output_dir, exist_ok=True)

    # Save filtered annotations to the output directory
    with open(os.path.join(output_dir, 'filtered_annotations.json'), 'w') as f:
        json.dump(filtered_annotations, f)

    print(f"Filtered annotations saved to {output_dir}")

# Example usage
auto_annotate_with_bboxes('path/to/annotations.json', class_id=1, output_dir='path/to/save_labels')
```

For more details, please visit our [Simple Utilities documentation](https://docs.ultralytics.com/usage/simple-utilities/).
